pkg/objects/config.go

```diff
@@ -208,7 +209,14 @@ func (cfg JobConfiguration) Compile(name string, swift SwiftLocation) (job *Job,
 	if cfg.Match.NotOlderThan != nil {
 		_, isSwiftSource := cfg.Source.src.(*SwiftLocation)
 		if !isSwiftSource {
-			errors = append(errors, fmt.Errorf("invalid value for %s.match.not_older_than: this option is only supported for source type %t", name, cfg.Source.src))
+			errors = append(errors, fmt.Errorf("invalid value for %s.match.not_older_than: this option is not supported for source type %T", name, cfg.Source.src))
+		}
+	}
+
+	if cfg.Match.SimplisticComparison != nil {
+		if cfg.Source.src == cfg.Source.src.(*DebianSource) ||
+			cfg.Source.src == cfg.Source.src.(*YumSource) {
+			errors = append(errors, fmt.Errorf("invalid value for %s.match.simplistic_comparsion: this option is not supported for source type %T", name, cfg.Source.src))
```

Comment:
Since we might add further specialized sources in this future, this would be safer as a whitelist rather than a blacklist, i.e. `if !isUrlSource && !isSwiftSource { complain() }`.